tests/test_main.py

```diff
@@ -0,0 +1,54 @@
+#! /usr/bin/env python
+
+import unittest
+import numpy as np
+from mecode import G
+
+
+g = G(
+    print_lines=False,
+    aerotech_include=False,
+    direct_write=True,
+    direct_write_mode='serial',
+    #printer_port="/dev/tty.usbmodem1411",
+    printer_port="COM6",
+)
+
+
+def read_profilometer(samples=1):
+    samples = np.floor(np.log2(samples))
```

Comment:
Can you help me understand how `samples` is calculated here? Why is the floor function and log-base-2 function needed?